src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeConstructorInfo.CoreCLR.cs

```diff
@@ -118,6 +119,13 @@ public override string ToString()
 
             return m_toString;
         }
+
+        public override bool Equals(object? obj) =>
+            ReferenceEquals(this, obj) ||
+            (MetadataUpdater.IsSupported && CacheEquals(obj));
```

Comment:
I think CacheEquals needs to check the declaring type for this to work. Otherwise, you can get false positives for different generic instantiations of ConstructorInfo